Modules/core/Controller/CoretilesController.php

```diff
@@ -99,4 +109,91 @@ public function docAction(){
         ));
     }
 
+
+    /**
+     * Distinctly list spaces:
+     * - of which user is member
+     * - in which user has a pending request to join
+     * - of which user is admin
+     * 
+     * @return array of arrays: [userSpaceIds, userPendingSpaceIds, SpacesUserIsAdminOf]
+     */
+    public function getUserSpaces() {
+        $modelSpacePending = new CorePendingAccount();
+        $data = $modelSpacePending->getSpaceIdsForPending($_SESSION["id_user"]);
+        $userPendingSpaceIds = array();
+
+        if ($data && count($data) > 0) {
+            foreach ($data as $space) {
+                array_push($userPendingSpaceIds, $space["id_space"]);
+            }
+        }
+
+        $modelSpaceUser = new CoreSpaceUser();
+        $data = $modelSpaceUser->getUserSpaceInfo($_SESSION["id_user"]);
+        $userSpaceIds = array();
+        $spacesUserIsAdminOf = array();
+
+        if ($data && count($data) > 0) {
+            foreach ($data as $space) {
+                array_push($userSpaceIds, $space["id_space"]);
+                if ($space["status"] === "4") {
+                    array_push($spacesUserIsAdminOf, $space["id_space"]);
+                }
+            }
+        }
+        $result = array(
+            "userSpaceIds" => $userSpaceIds,
+            "userPendingSpaceIds" => $userPendingSpaceIds,
+            "spacesUserIsAdminOf" => $spacesUserIsAdminOf
+        );
+
+        return $result;
+    }
+
+    /**
+     * 
+     * Manage actions resulting from user request to join or leave a space
+     *
+     * @param int $id_space
+     * @param bool $isMemberOfSpace
+     */
+    public function joinSpaceAction($id_space, $isMemberOfSpace) {
+        Configuration::getLogger()->debug("in DELETING", ["isMemberOfSpace: " => $isMemberOfSpace, "id_space: " => $id_space]);
```

Comment:
bad comment descrption

Reply:
Forgot a personal debug message. I remove it